### PR TITLE
Add apcupsd input plugin

### LINKED ISSUE #2877: Create an APCUPCd Plug-in
## Feature Request
Create a native APCUPCd plug-in for Telegraf.
http://www.apcupsd.org/

### Current behavior:
You can leverage the exec plug-in to gather stats from APC/UPC today...and example of that can be found here:
https://rmoff.net/2016/07/18/configuring-ups-apcupsd/
https://bitbucket.org/snippets/wnasich/7Kg89/telegraf-input-for-apc-ups-status-using

### Desired behavior:
Create a native plug-in which allows for gathering of stats from the daemon.

### Use case: 
The large and growing distribution of uninterruptible power supplies having stats which monitor these devices is important.

### PR DESCRIPTION
**Pull request by glinton**
Continues #2552
Resolves #2877

### Required for all PRs:

- [ ] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/apcupsd/apcupsd.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		}
+
+		tags := map[string]string{
+			"serial":   status.SerialNumber,
+			"ups_name": status.UPSName,
+			"status":   status.Status,
+		}
+
+		fields := map[string]interface{}{
+			"status_online":          boolToInt(status.Status == "ONLINE"), // todo: revisit
```
Comment (danielnelson) [inline plugins/inputs/apcupsd/apcupsd.go]: https://github.com/influxdata/telegraf/pull/2552/files#r248819823

File: plugins/inputs/apcupsd/apcupsd_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			{
+				servers: []string{"tcp://" + lAddr},
+				err:     false,
+			},
+		}
+
+		acc testutil.Accumulator
+	)
+
+	for _, test := range tests {
```
Comment (danielnelson) [inline plugins/inputs/apcupsd/apcupsd_test.go]: Cleanup tests in general and convert to table tests

File: plugins/inputs/apcupsd/apcupsd.go (hunk 1 above)
Comment (glinton) [reply plugins/inputs/apcupsd/apcupsd.go]: Due to the the lack of documentation of potential status flags, I feel just adding `status_flags` is a sufficient alternative.

The examples [in the source](https://sourceforge.net/projects/apcupsd/files/apcupsd%20-%20Stable/3.14.14/) differ as well - one ups example has a status of `ONLINE` with a statflag of `0x07000008`.

Comment (danielnelson) [reply plugins/inputs/apcupsd/apcupsd.go]: Okay, make the status_flags an uint64 so we can use the bitwise operators in InfluxQL. 
https://docs.influxdata.com/influxdb/v1.7/query_language/math_operators/#bitwise-and

I think the flags are defined in `include/defines.h`, with the extended bit values there are probably too many of them to handle anyway.  Plugin users can use the regex tag matching to find the status they are looking for.
```
/* bit values for APC UPS Status Byte (ups->Status) */
#define UPS_calibration   0x00000001
#define UPS_trim          0x00000002
#define UPS_boost         0x00000004
#define UPS_online        0x00000008
#define UPS_onbatt        0x00000010
#define UPS_overload      0x00000020
#define UPS_battlow       0x00000040
#define UPS_replacebatt   0x00000080

/* Extended bit values added by apcupsd */
#define UPS_commlost      0x00000100    /* Communications with UPS lost */
#define UPS_shutdown      0x00000200    /* Shutdown in progress */
#define UPS_slave         0x00000400    /* Set if this is a slave */
#define UPS_slavedown     0x00000800    /* Slave not responding */
#define UPS_onbatt_msg    0x00020000    /* Set when UPS_ONBATT message is sent */
#define UPS_fastpoll      0x00040000    /* Set on power failure to poll faster */
#define UPS_shut_load     0x00080000    /* Set when BatLoad <= percent */
#define UPS_shut_btime    0x00100000    /* Set when time on batts > maxtime */
#define UPS_shut_ltime    0x00200000    /* Set when TimeLeft <= runtime */
#define UPS_shut_emerg    0x00400000    /* Set when battery power has failed */
#define UPS_shut_remote   0x00800000    /* Set when remote shutdown */
#define UPS_plugged       0x01000000    /* Set if computer is plugged into UPS */
#define UPS_battpresent   0x04000000    /* Indicates if battery is connected */
```

**Pull request merged**